doc/sphinxext/related_software.py

```diff
@@ -0,0 +1,226 @@
+"""Create a list of related software.
+
+To add a package to the list:
+
+1. Add it to the MNE-installers if possible, and it will automatically appear.
+2. If it's on PyPI and not in the MNE-installers, add it to the PYPI_PACKAGES set.
+3. If it's not on PyPI, add it to the MANUAL_PACKAGES dictionary.
+"""
+
+import functools
+import importlib.metadata
+import os
+import pathlib
+import urllib.request
+
+import joblib
+from docutils import nodes
+from docutils.parsers.rst import Directive
+from sphinx.errors import ExtensionError
+from sphinx.util.display import status_iterator
+
+# If a package is in MNE-Installers, it will be automatically added!
+
+# If it's available on PyPI, add it to this set:
+PYPI_PACKAGES = {
+    "alphaCSC",
+    "conpy",
+    "meggie",
+    "niseq",
+    "sesameeg",
+}
+
+# If it's not available on PyPI, add it to this dict:
+MANUAL_PACKAGES = {
+    # TODO: These packages are not pip-installable as of 2024/07/17, so we have to
+    # manually populate them -- should open issues on their package repos.
+    "best-python": {
+        "Home-page": "https://github.com/multifunkim/best-python",
+        "Summary": "The goal of this project is to provide a way to use the best-brainstorm Matlab solvers in Python, compatible with MNE-Python.",  # noqa: E501
+    },
+    "mne-hcp": {
+        "Home-page": "https://github.com/mne-tools/mne-hcp",
+        "Summary": "We provide Python tools for seamless integration of MEG data from the Human Connectome Project into the Python ecosystem",  # noqa: E501
+    },
+    "posthoc": {
+        "Home-page": "https://users.aalto.fi/~vanvlm1/posthoc/python",
+        "Summary": "post-hoc modification of linear models",
+    },
+    # This package does not provide wheels, so don't force CircleCI to build it.
+    # If it eventually provides binary wheels we could add it to doc_related in
+    # pyproject.toml and remove from here.
+    "eelbrain": {
+        "Home-page": "https://eelbrain.readthedocs.io/en/stable/",
+        "Summary": "Open-source Python toolkit for MEG and EEG data analysis.",
+    },
+    # mne-kit-gui requires mayavi (ugh)
+    "mne-kit-gui": {
+        "Home-page": "https://github.com/mne-tools/mne-kit-gui",
+        "Summary": "A module for KIT MEG coregistration.",
+    },
+    # fsleyes requires wxpython, which needs to build
+    "fsleyes": {
+        "Home-page": "https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FSLeyes",
+        "Summary": "FSLeyes is the FSL image viewer.",
+    },
+    # dcm2niix must be built from source
+    "dcm2niix": {
+        "Home-page": "https://github.com/rordenlab/dcm2niix",
+        "Summary": "DICOM to NIfTI converter",
+    },
+    # TODO: mnelab forces PySide6, it can be added to doc_related when we use PySide6
+    # for doc building. Also its package does not set the Home-page property.
```

Comment:
```suggestion
    # TODO: mnelab forces PySide6, it can be added to `tools/circleci_dependencies.sh`
    # when we use PySide6 for doc building. Also its package does not set the Home-page
    # property.
```